packages/mobx/src/core/atom.ts

```diff
@@ -66,6 +67,9 @@ export class Atom implements IAtom {
     public reportChanged() {
         startBatch()
         propagateChanged(this)
+        // We could update state version only at the end of batch,
+        // but we would still have to switch some global flag here to signal a change.
+        globalState.stateVersion = Symbol()
```

Comment:
Why not increase a global number? Symbol() is probably a lot more expensive (both mem and cpu), given that this is probably hit very often

Reply:
What makes you say `Symbol()` is more expensive?

Reply:
> Why not increase a global number?

Becomes non-unique at some point. Dunno how much of an issue it would be in practice. For reference https://github.com/mobxjs/mobx/pull/2982

Reply:
Changed to number for now. The only thing I am concerned is that if this should ever become a problem, nobody would probably ever notice anyway... not during development at least.